lib/Conversion/PolyToStandard/PolyToStandard.cpp

```diff
@@ -107,6 +115,71 @@ struct ConvertToTensor : public OpConversionPattern<ToTensorOp> {
   }
 };
 
+struct ConvertMonomial : public OpConversionPattern<MonomialOp> {
+  ConvertMonomial(mlir::MLIRContext *context)
+      : OpConversionPattern<MonomialOp>(context) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      MonomialOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
+    auto polyType = cast<PolyType>(op.getResult().getType());
+    auto tensorType =
+        cast<RankedTensorType>(typeConverter->convertType(polyType));
+    auto tensor = b.create<arith::ConstantOp>(DenseElementsAttr::get(
+        tensorType, b.getIntegerAttr(tensorType.getElementType(), 0)));
+    rewriter.replaceOpWithNewOp<tensor::InsertOp>(op, adaptor.getCoefficient(),
+                                                  tensor, adaptor.getDegree());
+    return success();
+  }
+};
+
+struct ConvertDegree : public OpConversionPattern<DegreeOp> {
+  ConvertDegree(mlir::MLIRContext *context)
+      : OpConversionPattern<DegreeOp>(context) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  // FIXME: Is it worth implementing this as a function call as well?
+  LogicalResult matchAndRewrite(
+      DegreeOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
+    auto coeffs = adaptor.getInput();
+    auto tensorType = cast<RankedTensorType>(coeffs.getType());
+    auto c0 = b.create<arith::ConstantOp>(
+        b.getIntegerAttr(tensorType.getElementType(), 0));
+    auto c1 = b.create<arith::ConstantOp>(b.getIndexAttr(1));
+    auto initIndex = b.create<arith::ConstantOp>(
+        b.getIndexAttr(tensorType.getShape()[0] - 1));
+
+    rewriter.replaceOpWithNewOp<scf::WhileOp>(
+        op,
+        /*resultTypes=*/TypeRange{b.getIndexType()},
+        /*operands=*/ValueRange{initIndex.getResult()},
+        /*beforeBuilder=*/
+        [&](OpBuilder &nestedBuilder, Location nestedLoc, ValueRange args) {
+          Value index = args[0];
+          ImplicitLocOpBuilder b(nestedLoc, nestedBuilder);
+          auto coeff = b.create<tensor::ExtractOp>(coeffs, ValueRange{index});
+          auto cmpOp =
+              b.create<arith::CmpIOp>(arith::CmpIPredicate::sgt, coeff, c0);
```

Comment:
Small question for me: By using "sgt", do you interpret the unsigned coefficient as signed? Would that be an issue for large coeffs?

It's possible here I think to also use a "not equal to" op.

Reply:
Fixed.